test/fake-intake/fakeintake/server_test.go

```diff
@@ -0,0 +1,131 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fakeintake
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPOSTPayloads(t *testing.T) {
+	t.Run("should accept payloads on any route", func(t *testing.T) {
+		fi := NewFakeIntake()
+		defer fi.server.Close()
+
+		request, err := http.NewRequest(http.MethodPost, "/api/v2/series", strings.NewReader("totoro|5|tag:valid,owner:pducolin"))
+		assert.NoError(t, err, "Error creating POST request")
+		response := httptest.NewRecorder()
+
+		fi.postPayload(response, request)
+
+		expectedResponse := postPayloadResponse{
+			Errors: []string{},
+		}
+
+		actualResponse := postPayloadResponse{}
+		body, err := io.ReadAll(response.Body)
+		assert.NoError(t, err, "Error reading response")
+		json.Unmarshal(body, &actualResponse)
+
+		assert.Equal(t, expectedResponse, actualResponse, "unexpected response")
+	})
+
+	t.Run("should not accept GET requests on any route", func(t *testing.T) {
+		fi := NewFakeIntake()
+		defer fi.server.Close()
+
+		request, err := http.NewRequest(http.MethodGet, "/api/v2/series", nil)
+		assert.NoError(t, err, "Error creating GET request")
+		response := httptest.NewRecorder()
+
+		fi.postPayload(response, request)
+
+		expectedResponse := postPayloadResponse{
```

Comment:
suggestion: You can simplify the test by only checking the http code is `405`.

Reply:
No check on the body ? I would add a check for the status plus the expected error in the body, what do you think ?